redisson/src/main/java/org/redisson/RedissonMapCache.java

```diff
@@ -261,68 +287,93 @@ public V putIfAbsent(K key, V value, long ttl, TimeUnit ttlUnit, long maxIdleTim
+                            + "else "
+                                + "redis.call('zrem', KEYS[3], ARGV[5]); "
+                            + "end; "
+
+                            // last access time
+                            + "local maxSize = tonumber(ARGV[7]);" +
+                            "if maxSize ~= 0 then" +
+                            "    local currentTime = tonumber(ARGV[1]);" +
+                            "    local lastAccessTimeSetName = KEYS[5];" +
+                            "    redis.call('zadd', lastAccessTimeSetName, currentTime, ARGV[5]);" +
+                            "    local cacheSize = tonumber(redis.call('hlen', KEYS[1]));" +
+                            "    if cacheSize >= maxSize then" +
+                            "        local lruItems = redis.call('zrange', lastAccessTimeSetName, 0, cacheSize - maxSize);" +
+                            "        for index, lruItem in ipairs(lruItems) do" +
+                            "            if lruItem then" +
+                            "                local lruItemValue = redis.call('hget', KEYS[1], lruItem);" +
+                            "                redis.call('hdel', KEYS[1], lruItem);" +
+                            "                redis.call('zrem', KEYS[2], lruItem);" +
+                            "                redis.call('zrem', KEYS[3], lruItem);" +
+                            "                redis.call('zrem', lastAccessTimeSetName, lruItem);" +
+                            "                local removedChannelName = KEYS[6];" +
+                            "                local msg = struct.pack('Lc0Lc0', string.len(lruItem), lruItem, string.len(lruItemValue), lruItemValue);" +
+                            "                redis.call('publish', removedChannelName, msg);" +
+                            "            end;" +
+                            "        end" +
+                            "    end;" +
+                            "end;"
+
+                            // value
+                            + "local val = struct.pack('dLc0', tonumber(ARGV[4]), string.len(ARGV[6]), ARGV[6]); "
+                            + "redis.call('hset', KEYS[1], ARGV[5], val); "
+
+                            + "local msg = struct.pack('Lc0Lc0', string.len(ARGV[5]), ARGV[5], string.len(ARGV[6]), ARGV[6]); "
+                            + "redis.call('publish', KEYS[4], msg); "
+
+                            + "return nil;"
+                        + "else "
+                            + "local t, val = struct.unpack('dLc0', value); "
+                            + "redis.call('zadd', KEYS[3], t + ARGV[1], ARGV[5]); "
+                            + "return val;"
+                        + "end; ",
+                Arrays.<Object>asList(getName(key), getTimeoutSetNameByKey(key), getIdleSetNameByKey(key), getCreatedChannelNameByKey(key),
+                        getLastAccessTimeSetName(), getRemovedChannelName()),
```

Comment:
@johnou getLastAccessTimeSetNameByKey, getRemovedChannelNameByKey should be used instead